### PR TITLE
Issue 1119 - Show contact names and email adresses

### PR DESCRIPTION
**Pull request by MaximeMenotti**
Hi, 
has suggested the issue https://github.com/k9mail/k-9/issues/1119 I add an option to show both real name and email address.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/helper/MessageHelper.java [hunk 1]
```diff
@@ -179,7 +182,7 @@ public static CharSequence toFriendly(Address[] addresses, Contacts contacts) {
                 }
             }
         }
-
+        if(showCorrespondentNamesAndEmailAdresses) return (!TextUtils.isEmpty(address.getPersonal())) ? address.getPersonal() + " <" + address.getAddress() + ">" : address.getAddress();
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/helper/MessageHelper.java]: This line is really long. Also the logic is a bit convoluted - a return containing a ! and a ternary

**Comment by philipwhiuk** [comment]
The build failed because you broke a related test. Also it would be good to add some. What happens if `mShowCorrespondentNames` is false but `mShowCorrespondentNamesAndAddresses` is true.

To me it seems like it would be better changing `mShowCorrespondentNames` from a boolean to a three-value enum.

**Comment by MaximeMenotti** [comment]
Sorry for this mistake, how can I run the unit tests on my own ? I tried to build the project with the "gradlew.bat testDebug" and "gradlew.bat assembleDebug" commands but it seems that all tests passed...

File: k9mail/src/main/java/com/fsck/k9/helper/MessageHelper.java (hunk 1 above)
Comment (MaximeMenotti) [inline k9mail/src/main/java/com/fsck/k9/helper/MessageHelper.java]: Ok, I take note, it will be corrected in my futur commit.

**Comment by cketti** [comment]
Try `gradlew.bat clean testDebug`.

**Comment by cketti** [comment]
@philipwhiuk is right. The appearance of correspondents should be configured using only one setting. Both in code and in the user interface.

**Comment by MaximeMenotti** [comment]
I add the three value enum and I fix the unit test problem as expected. 
However I still have a problem. When the user choose between the three option the change is real but the interface is not refresh (I have to return out of the parameters to refresh the interface).

Can you see were I'm wrong ?

File: k9mail/src/main/java/com/fsck/k9/K9.java [hunk 2]
```diff
@@ -210,11 +210,17 @@
         WHEN_IN_LANDSCAPE
     }
 
+    public enum ContactViewMode {
+        EMAIL,
+        NAME,
+        BOTH
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/K9.java]: Without the context of the other enum values it's not immediately obvious what `BOTH` means. Please rename to `NAME_AND_EMAIL`.

File: k9mail/src/main/java/com/fsck/k9/K9.java [hunk 3]
```diff
@@ -553,17 +559,17 @@ private void broadcastIntent(String action, Account account, String folder, Mess
                     K9.this.sendBroadcast(intent);
                     if (K9.DEBUG)
                         Log.d(K9.LOG_TAG, "Broadcasted: action=" + action
-                              + " account=" + account.getDescription()
-                              + " folder=" + folder
-                              + " message uid=" + message.getUid()
-                             );
+                                        + " account=" + account.getDescription()
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/K9.java]: Please revert unrelated changes. There are a couple more of these whitespace changes.

**Comment by cketti** [comment]
I've thought a bit more about it and don't like the user experience of this approach. See #1119 for more details.
